lib/puppet/type/notify.rb

```diff
@@ -6,7 +6,7 @@ module Puppet
   Type.newtype(:notify) do
     @doc = "Sends an arbitrary message to the agent run-time log."
 
-    newproperty(:message) do
+    newproperty(:message, :idempotent => false) do
```

Comment:
FYI - I just threw `:idempotent => false` on some properties in existing types on the existing codebase, and ran the spec suite (just to see if anything would break).  I hoping this serves as a proxy for newer modules opting to use `:idempotent` running on an older Puppet where that concept doesn't exist.

````
diff --git a/lib/puppet/type/exec.rb b/lib/puppet/type/exec.rb
index f2526ba..11673e5 100644
--- a/lib/puppet/type/exec.rb
+++ b/lib/puppet/type/exec.rb
@@ -64,7 +64,7 @@ module Puppet
       @checks.keys
     end

-    newproperty(:returns, :array_matching => :all, :event => :executed_command) do |property|
+    newproperty(:returns, :array_matching => :all, :event => :executed_command, :idempotent => false) do |property|
       include Puppet::Util::Execution
       munge do |value|
         value.to_s
diff --git a/lib/puppet/type/file/content.rb b/lib/puppet/type/file/content.rb
index 614d9d7..9c98a58 100644
--- a/lib/puppet/type/file/content.rb
+++ b/lib/puppet/type/file/content.rb
@@ -6,7 +6,7 @@ require 'puppet/util/checksums'
 require 'puppet/type/file/data_sync.rb'

 module Puppet
-  Puppet::Type.type(:file).newproperty(:content) do
+  Puppet::Type.type(:file).newproperty(:content, :idempotent => false) do
     include Puppet::Util::Checksums
     include Puppet::DataSync

diff --git a/lib/puppet/type/package.rb b/lib/puppet/type/package.rb
index cab3c67..2b3e159 100644
--- a/lib/puppet/type/package.rb
+++ b/lib/puppet/type/package.rb
@@ -296,7 +296,7 @@ module Puppet
       [ [ /(.*)/m, [ [:name] ] ] ]
     end

-    newproperty(:package_settings, :required_features=>:package_settings) do
+    newproperty(:package_settings, :required_features=>:package_settings, :idempotent => false) do
       desc "Settings that can change the contents or configuration of a package.

         The formatting and effects of package_settings are provider-specific; any
diff --git a/lib/puppet/type/scheduled_task.rb b/lib/puppet/type/scheduled_task.rb
index 97215a2..1d6eb96 100644
--- a/lib/puppet/type/scheduled_task.rb
+++ b/lib/puppet/type/scheduled_task.rb
@@ -54,7 +54,7 @@ Puppet::Type.newtype(:scheduled_task) do
       should be specified as a space-separated string."
   end

-  newproperty(:user) do
+  newproperty(:user, :idempotent => false) do
     desc "The user to run the scheduled task as.  Please note that not
       all security configurations will allow running a scheduled task
       as 'SYSTEM', and saving the scheduled task under these
diff --git a/lib/puppet/type/service.rb b/lib/puppet/type/service.rb
index 1f8d266..8a5508c 100644
--- a/lib/puppet/type/service.rb
+++ b/lib/puppet/type/service.rb
@@ -122,7 +122,7 @@ module Puppet
       end
     end

-    newproperty(:flags, :required_features => :flaggable) do
+    newproperty(:flags, :required_features => :flaggable, :idempotent => false) do
       desc "Specify a string of flags to pass to the startup script."
     end

diff --git a/lib/puppet/type/user.rb b/lib/puppet/type/user.rb
index 3c1156e..78c9f36 100644
--- a/lib/puppet/type/user.rb
+++ b/lib/puppet/type/user.rb
@@ -98,7 +98,7 @@ module Puppet
       end
     end

-    newproperty(:home) do
+    newproperty(:home, :idempotent => false) do
       desc "The home directory of the user.  The directory must be created
         separately and is not currently checked for existence."
     end
```

All the specs passed on Windows:

```
Finished in 11 minutes 58 seconds (files took 15.14 seconds to load)
21219 examples, 0 failures, 62 pending
```